beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/publisher/AbstractBlockPublisher.java

```diff
@@ -62,47 +68,54 @@ public SafeFuture<SendSignedBlockResult> sendSignedBlock(
-                  signedBlock, blobSidecars, broadcastValidationLevel, blockPublishingPerformance);
-            })
+            // creating blob sidecars after unblinding the block to ensure in the blinded flow we
+            // will have the cached builder payload
+            signedBlock ->
+                gossipAndImportUnblindedSignedBlockAndBlobSidecars(
+                    signedBlock,
+                    Suppliers.memoize(
+                        () ->
+                            blockFactory.createBlobSidecars(
+                                blockContainer, blockPublishingPerformance)),
+                    broadcastValidationLevel,
+                    blockPublishingPerformance))
         .thenCompose(result -> calculateResult(blockContainer, result, blockPublishingPerformance));
   }
 
   private SafeFuture<BlockImportAndBroadcastValidationResults>
       gossipAndImportUnblindedSignedBlockAndBlobSidecars(
           final SignedBeaconBlock block,
-          final List<BlobSidecar> blobSidecars,
+          final Supplier<List<BlobSidecar>> blobSidecars,
           final BroadcastValidationLevel broadcastValidationLevel,
           final BlockPublishingPerformance blockPublishingPerformance) {
 
     if (broadcastValidationLevel == BroadcastValidationLevel.NOT_REQUIRED) {
       // when broadcast validation is disabled, we can publish the block (and blob sidecars)
       // immediately and then import
-      publishBlockAndBlobSidecars(block, blobSidecars, blockPublishingPerformance);
-      return importBlockAndBlobSidecars(
-          block, blobSidecars, broadcastValidationLevel, blockPublishingPerformance);
+      publishBlock(block, blockPublishingPerformance)
+          .always(() -> publishBlobSidecars(blobSidecars.get(), blockPublishingPerformance));
+
+      importBlobSidecars(blobSidecars.get(), blockPublishingPerformance);
+      return importBlock(block, broadcastValidationLevel, blockPublishingPerformance);
     }
 
     // when broadcast validation is enabled, we need to wait for the validation to complete before
     // publishing the block (and blob sidecars)
 
     final SafeFuture<BlockImportAndBroadcastValidationResults>
         blockImportAndBroadcastValidationResults =
-            importBlockAndBlobSidecars(
-                block, blobSidecars, broadcastValidationLevel, blockPublishingPerformance);
+            importBlock(block, broadcastValidationLevel, blockPublishingPerformance);
 
     blockImportAndBroadcastValidationResults
         .thenCompose(BlockImportAndBroadcastValidationResults::broadcastValidationResult)
         .thenAccept(
             broadcastValidationResult -> {
               if (broadcastValidationResult == BroadcastValidationResult.SUCCESS) {
-                publishBlockAndBlobSidecars(block, blobSidecars, blockPublishingPerformance);
+                publishBlock(block, blockPublishingPerformance)
+                    .always(
```

Comment:
Actually, why do we wait for block to finish publishing again and not just send both across async?

Reply:
that's a strategy that has been discussed in several places. The reason are essentially help solo stakers with low up bandwidth to reduce latency in block propagation. With the adoption of `engine_getBlobs` blobs can be retrieved once the block is known, thus it is good to give priority to the block over the blobs.

I'll put this under a feature flag.